test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -21,9 +21,25 @@
  * Unit tests for the solidity expression compiler, testing the behaviour of the code.
  */
 
+#include <functional>
 #include <string>
 #include <tuple>
 #include <boost/test/unit_test.hpp>
+
+#if defined(__GNUC__)
+#pragma warning(push)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+#endif // defined(__GNUC__)
+
+#include <boost/thread.hpp>
```

Comment:
Where is this needed? There is a sleep somewhere, but that should be taken from stdc++11, not from boost.